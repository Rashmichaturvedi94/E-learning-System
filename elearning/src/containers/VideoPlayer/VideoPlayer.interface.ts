export interface VideoPlayerProps {
  // TODO: add required props below
  myProp?: boolean;
}

export type SelectedTextTrack = {
  type: string;
  value: string | number;
};

export type TextTrack = {
  title?: string;
  language: string;
  index: number;
};
