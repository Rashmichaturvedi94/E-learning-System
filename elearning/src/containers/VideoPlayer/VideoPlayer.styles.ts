import styled from 'styled-components/native';
import { Animated, Text, View, StyleSheet } from 'react-native';
import { Icon } from 'react-native-elements';
import * as Progress from 'react-native-progress';

export const Container = styled(View)`
  flex: 1;
  top: 0px;
  background-color: #000000;
`;

export const TopContainer = styled(View)`
  flex-direction: row;
  align-content: center;
  justify-content: space-between;
  margin: 0px 16px;
  margin-top: 24px;
`;

export const PlayOptionContainer = styled(View)`
  flex-direction: row;
  justify-content: center;
  align-items: center;
`;

export const PlayButton = styled(Icon).attrs((props) => ({
  name: props.name,
  color: '#ffffff',
  size: 52,
}))`
  margin: 0px 32px;
`;

export const OptionsContainer = styled(View)`
  flex-direction: row;
  justify-content: space-between;
`;

export const CloseIcon = styled(Icon).attrs({
  name: 'closed-caption',
  color: '#fff',
})`
  height: 24px;
  width: 24px;
  margin-right: 56px;
`;

export const CCIcon = styled(Icon).attrs({
  name: 'close',
  color: '#fff',
})`
  height: 24px;
  width: 24px;
  margin-right: 56px;
`;

export const TitleContainer = styled(View)`
  flex-direction: row;
  margin-top: 8px;
`;

export const ControlsContainer = styled(View)<{ hidden: boolean }>`
  flex-direction: column;
  bottom: 0px;
  left: 0px;
  right: 0px;
  justify-content: space-between;
  height: 100%;
  z-index: ${({ hidden }) => (hidden ? -1 : 1)};
  opacity: ${({ hidden }) => (hidden ? 0 : 1)};
`;

export const ProgressContainer = styled(View)`
  height: 48px;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  padding: 0px 10px;
  margin: 0px 48px;
  margin-bottom: 24px;
`;
export const SeekBar = styled(Progress.Bar).attrs({
  color: 'white',
  unfilledColor: 'rgba(255, 255, 255, 0.5)',
})`
  color: #fff;
  border-color: rgba(255, 255, 255, 0);
  height: 3px;
`;

export const Duration = styled(Text)`
  color: #fff;
  opacity: 0.5;
  font-weight: 700;
  font-size: 14px;
  margin-left: 15px;
`;

export const CurrentProgress = styled(Animated.View)`
  flex-direction: column;
  position: absolute;
  bottom: -8px;
`;

export const ProgressKnob = styled(View)`
  background-color: #fff;
  width: 18px;
  height: 18px;
  border-radius: 9px;
`;

export const styles = StyleSheet.create({
  backgroundVideo: {
    position: 'absolute',
    top: 0,
    left: 0,
    bottom: 0,
    right: 0,
  },
});
