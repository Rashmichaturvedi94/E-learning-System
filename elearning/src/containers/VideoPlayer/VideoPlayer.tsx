import React, { FC, useRef, useEffect, useState } from 'react';
import Video from 'react-native-video';
import { useNavigation, useRoute } from '@react-navigation/native';
import {
  Animated,
  Dimensions,
  Easing,
  TouchableOpacity,
  TouchableWithoutFeedback,
  View,
} from 'react-native';
import Orientation from 'react-native-orientation-locker';
import { Icon } from 'react-native-elements';
import { CollectionKeys, getUser, secondsToDuration } from 'utils/utils';
import firestore from '@react-native-firebase/firestore';
import { useActionSheet } from '@expo/react-native-action-sheet';
import ISO6391 from 'iso-639-1';
import {
  TapGestureHandler,
  PanGestureHandler,
} from 'react-native-gesture-handler';
import {
  VideoPlayerProps,
  TextTrack,
  SelectedTextTrack,
} from './VideoPlayer.interface';
import {
  Container,
  ControlsContainer,
  Duration,
  OptionsContainer,
  PlayButton,
  PlayOptionContainer,
  ProgressContainer,
  SeekBar,
  CloseIcon,
  TopContainer,
  CCIcon,
  styles,
  CurrentProgress,
  ProgressKnob,
  CurrentTime,
} from './VideoPlayer.styles';

export const VideoPlayer: FC<VideoPlayerProps> = () => {
  const progressBarWidth = Dimensions.get('window').width * 0.8;
  const { params } = useRoute();
  const { score } = params;
  const xPosition = useRef(new Animated.Value(0)).current;
  const [paused, setPause] = useState(false);
  const [progress, setProgress] = useState(0);
  const [duration, setDuration] = useState(0);
  const [currentTime, setCurrentTime] = useState(0);
  const [textTracks, setTextTracks] = useState<TextTrack[]>([]);
  const [selectedCC, setSelectedCC] = useState<SelectedTextTrack>();
  const player = useRef<Video | null>(null);
  const navigation = useNavigation();
  const [overlayHidden, setOverlayHidden] = useState(false);
  const { showActionSheetWithOptions } = useActionSheet();
  const handleOnLoad = (meta: { duration: React.SetStateAction<number> }) => {
    setDuration(meta.duration);
    // @ts-ignore
    if (meta.textTracks instanceof Array) {
      // @ts-ignore
      setTextTracks(meta.textTracks as TextTrack[]);
    }
  };
  const handleOverlayPress = () => {
    setOverlayHidden((hidden) => !hidden);
  };
  const handleOnProgress = (progress1: { currentTime: number }) => {
    setCurrentTime(progress1.currentTime);
    setProgress(progress1.currentTime / duration);
    xPosition.setValue(progress1.currentTime / duration);
  };

  const updateScore = (user: User) => {
    const userRef = firestore().collection(CollectionKeys.USER).doc(user.uid);
    userRef.update({
      score: firestore.FieldValue.increment(Number(score)),
    });
  };

  const handleOnEnd = () => {
    setPause(true);
    getUser().then((user) => {
      updateScore(user);
    });
  };

  const handlePlayButton = () => {
    if (progress >= 1) {
      player.current?.seek(0);
    }
    setPause((pause) => !pause);
  };
  const handleOnReversePress = () => {
    player.current?.seek(currentTime - 10);
  };
  const handleOnForwardPress = () => {
    player.current?.seek(currentTime + 10);
  };
  const handleOnProgressPress = (e: { nativeEvent: { x: number } }) => {
    const position = e.nativeEvent.x;
    const seekTo = (position / progressBarWidth) * duration;
    player.current?.seek(seekTo);
  };
  const handleCCPress = () => {
    const titles = textTracks?.map((item) => {
      if (item.language !== undefined) {
        if (item.language.length < 3) {
          return ISO6391.getName(item.language);
        }
        return item.language;
      }
      return '';
    });
    const options = ['None', ...titles];

    showActionSheetWithOptions(
      {
        options,
      },
      (buttonIndex) => {
        if (buttonIndex === 0) {
          setSelectedCC({ type: 'language', value: '' });
        } else {
          const item = textTracks[buttonIndex - 1];
          setSelectedCC({ type: 'language', value: item.language });
        }
      },
    );
  };
  useEffect(() => {
    Orientation.lockToLandscapeLeft();
    const parent = navigation.dangerouslyGetParent();
    if (parent) {
      parent.setOptions({
        tabBarVisible: false,
      });
    }
    return () => {
      if (parent) {
        parent.setOptions({
          tabBarVisible: true,
        });
        Orientation.lockToPortrait();
      }
    };
  }, [navigation]);
  const interpolatedX = xPosition.interpolate({
    inputRange: [0, 1],
    outputRange: [0, progressBarWidth],
    easing: Easing.linear,
  });
  return (
    <Container>
      <TouchableWithoutFeedback onPress={handleOverlayPress}>
        <View>
          <Video
            source={{
              uri:
                'https://devstreaming-cdn.apple.com/videos/streaming/examples/bipbop_16x9/bipbop_16x9_variant.m3u8',
            }}
            style={styles.backgroundVideo}
            ref={player}
            paused={paused}
            resizeMode="contain"
            onLoad={handleOnLoad}
            onProgress={handleOnProgress}
            onEnd={handleOnEnd}
            selectedTextTrack={selectedCC}
          />
          <ControlsContainer hidden={overlayHidden}>
            <TopContainer>
              <OptionsContainer>
                <TouchableOpacity onPress={handleCCPress}>
                  <CloseIcon />
                </TouchableOpacity>
                <TouchableWithoutFeedback
                  onPress={() => {
                    navigation.goBack();
                  }}
                >
                  <CCIcon />
                </TouchableWithoutFeedback>
              </OptionsContainer>
            </TopContainer>
            <PlayOptionContainer>
              <TouchableOpacity onPress={handleOnReversePress}>
                <Icon name="history" size={30} color="#fff" />
              </TouchableOpacity>
              <TouchableWithoutFeedback onPress={handlePlayButton}>
                <PlayButton name={paused ? 'play-arrow' : 'pause'} />
              </TouchableWithoutFeedback>
              <TouchableOpacity onPress={handleOnForwardPress}>
                <Icon name="update" size={30} color="#fff" />
              </TouchableOpacity>
            </PlayOptionContainer>
            {!overlayHidden && (
              <ProgressContainer
                onStartShouldSetResponder={() => true}
                onTouchEnd={(e) => {
                  e.stopPropagation();
                }}
              >
                <PanGestureHandler onHandlerStateChange={handleOnProgressPress}>
                  <TapGestureHandler
                    onHandlerStateChange={handleOnProgressPress}
                  >
                    <View style={{ width: progressBarWidth }}>
                      <SeekBar progress={progress} width={progressBarWidth} />
                      <CurrentProgress
                        style={{
                          transform: [{ translateX: interpolatedX }],
                          width: progressBarWidth,
                        }}
                      >
                        <CurrentTime>
                          {secondsToDuration(currentTime ?? 0)}
                        </CurrentTime>
                        <ProgressKnob />
                      </CurrentProgress>
                    </View>
                  </TapGestureHandler>
                </PanGestureHandler>
                <Duration>{secondsToDuration(duration)}</Duration>
              </ProgressContainer>
            )}
          </ControlsContainer>
        </View>
      </TouchableWithoutFeedback>
    </Container>
  );
};
