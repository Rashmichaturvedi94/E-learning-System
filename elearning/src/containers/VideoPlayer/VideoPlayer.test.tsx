import React from 'react';
import { render } from '@testing-library/react-native';
import { createStackNavigator } from '@react-navigation/stack';
import { NavigationContainer } from '@react-navigation/native';
import { VideoPlayer } from './VideoPlayer';

const Stack = createStackNavigator();

describe('containers/VideoPlayer', () => {
  it('should render', () => {
    const component = render(
      <NavigationContainer>
        <Stack.Navigator
          initialRouteName="VideoPlayer"
          screenOptions={{
            headerBackTitleVisible: false,
          }}
        >
          <Stack.Screen
            name="VideoPlayer"
            component={VideoPlayer}
            initialParams={{ score: 5 }}
            options={{ headerShown: false }}
          />
        </Stack.Navigator>
      </NavigationContainer>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
