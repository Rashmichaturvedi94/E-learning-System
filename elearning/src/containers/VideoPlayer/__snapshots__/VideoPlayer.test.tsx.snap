// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/VideoPlayer should render 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <RNSScreenContainer
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <RNSScreen
        activityState={2}
        forwardedRef={[Function]}
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "opacity": 1,
            }
          }
        />
        <View
          accessibilityElementsHidden={false}
          closing={false}
          gestureVelocityImpact={0.3}
          importantForAccessibility="auto"
          onClose={[Function]}
          onGestureBegin={[Function]}
          onGestureCanceled={[Function]}
          onGestureEnd={[Function]}
          onOpen={[Function]}
          onTransition={[Function]}
          pointerEvents="box-none"
          style={
            Array [
              Object {
                "overflow": undefined,
              },
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
            ]
          }
          transitionSpec={
            Object {
              "close": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
              "open": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
            }
          }
        >
          <View
            needsOffscreenAlphaCompositing={false}
            pointerEvents="box-none"
            style={
              Object {
                "flex": 1,
                "marginTop": 0,
              }
            }
          >
            <View
              collapsable={false}
              forwardedRef={[Function]}
              onGestureHandlerEvent={[Function]}
              onGestureHandlerStateChange={[Function]}
              style={
                Object {
                  "flex": 1,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateX": 0,
                    },
                  ],
                }
              }
            >
              <View
                pointerEvents="none"
                style={
                  Object {
                    "backgroundColor": "rgb(242, 242, 242)",
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "shadowColor": "#000",
                    "shadowOffset": Object {
                      "height": 1,
                      "width": -1,
                    },
                    "shadowOpacity": 0.3,
                    "shadowRadius": 5,
                    "top": 0,
                    "width": 3,
                  }
                }
              />
              <View
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "overflow": "hidden",
                    },
                    Array [
                      Object {
                        "backgroundColor": "rgb(242, 242, 242)",
                      },
                      undefined,
                    ],
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                      "flexDirection": "column-reverse",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  >
                    <View
                      style={
                        Array [
                          Object {
                            "backgroundColor": "#000000",
                            "flexBasis": 0,
                            "flexGrow": 1,
                            "flexShrink": 1,
                            "top": 0,
                          },
                        ]
                      }
                    >
                      <View
                        accessible={true}
                        focusable={true}
                        onClick={[Function]}
                        onResponderGrant={[Function]}
                        onResponderMove={[Function]}
                        onResponderRelease={[Function]}
                        onResponderTerminate={[Function]}
                        onResponderTerminationRequest={[Function]}
                        onStartShouldSetResponder={[Function]}
                      >
                        <Video
                          onEnd={[Function]}
                          onLoad={[Function]}
                          onProgress={[Function]}
                          paused={false}
                          resizeMode="contain"
                          source={
                            Object {
                              "uri": "https://devstreaming-cdn.apple.com/videos/streaming/examples/bipbop_16x9/bipbop_16x9_variant.m3u8",
                            }
                          }
                          style={
                            Object {
                              "bottom": 0,
                              "left": 0,
                              "position": "absolute",
                              "right": 0,
                              "top": 0,
                            }
                          }
                        />
                        <View
                          hidden={false}
                          style={
                            Array [
                              Object {
                                "bottom": 0,
                                "flexDirection": "column",
                                "height": "100%",
                                "justifyContent": "space-between",
                                "left": 0,
                                "opacity": 1,
                                "right": 0,
                                "zIndex": 1,
                              },
                            ]
                          }
                        >
                          <View
                            style={
                              Array [
                                Object {
                                  "alignContent": "center",
                                  "flexDirection": "row-reverse",
                                  "justifyContent": "space-between",
                                  "marginBottom": 0,
                                  "marginLeft": 16,
                                  "marginRight": 16,
                                  "marginTop": 24,
                                },
                              ]
                            }
                          >
                            <View
                              style={
                                Array [
                                  Object {
                                    "flexDirection": "row",
                                    "justifyContent": "space-between",
                                    "marginBottom": 8,
                                    "marginLeft": 16,
                                    "marginRight": 16,
                                    "marginTop": 8,
                                  },
                                ]
                              }
                            >
                              <View
                                accessible={true}
                                focusable={true}
                                onClick={[Function]}
                                onResponderGrant={[Function]}
                                onResponderMove={[Function]}
                                onResponderRelease={[Function]}
                                onResponderTerminate={[Function]}
                                onResponderTerminationRequest={[Function]}
                                onStartShouldSetResponder={[Function]}
                                style={
                                  Object {
                                    "opacity": 1,
                                  }
                                }
                              >
                                <View
                                  style={
                                    Object {
                                      "overflow": "hidden",
                                    }
                                  }
                                >
                                  <View
                                    style={
                                      Array [
                                        Object {
                                          "height": 24,
                                          "marginBottom": 0,
                                          "marginLeft": 8,
                                          "marginRight": 8,
                                          "marginTop": 0,
                                          "width": 24,
                                        },
                                      ]
                                    }
                                  >
                                    <View
                                      style={
                                        Object {
                                          "alignItems": "center",
                                          "backgroundColor": "transparent",
                                          "justifyContent": "center",
                                        }
                                      }
                                    >
                                      <Text
                                        allowFontScaling={false}
                                        style={
                                          Array [
                                            Object {
                                              "color": "#fff",
                                              "fontSize": 24,
                                            },
                                            Object {
                                              "backgroundColor": "transparent",
                                            },
                                            Object {
                                              "fontFamily": "Material Icons",
                                              "fontStyle": "normal",
                                              "fontWeight": "normal",
                                            },
                                            Object {},
                                          ]
                                        }
                                        testID="iconIcon"
                                      >
                                        
                                      </Text>
                                    </View>
                                  </View>
                                </View>
                              </View>
                              <View
                                style={
                                  Object {
                                    "overflow": "hidden",
                                  }
                                }
                              >
                                <View
                                  accessible={true}
                                  focusable={true}
                                  onClick={[Function]}
                                  onResponderGrant={[Function]}
                                  onResponderMove={[Function]}
                                  onResponderRelease={[Function]}
                                  onResponderTerminate={[Function]}
                                  onResponderTerminationRequest={[Function]}
                                  onStartShouldSetResponder={[Function]}
                                  style={
                                    Array [
                                      Object {
                                        "height": 24,
                                        "marginBottom": 0,
                                        "marginLeft": 8,
                                        "marginRight": 8,
                                        "marginTop": 0,
                                        "width": 24,
                                      },
                                    ]
                                  }
                                >
                                  <View
                                    style={
                                      Object {
                                        "alignItems": "center",
                                        "backgroundColor": "transparent",
                                        "justifyContent": "center",
                                      }
                                    }
                                  >
                                    <Text
                                      allowFontScaling={false}
                                      style={
                                        Array [
                                          Object {
                                            "color": "#fff",
                                            "fontSize": 24,
                                          },
                                          Object {
                                            "backgroundColor": "transparent",
                                          },
                                          Object {
                                            "fontFamily": "Material Icons",
                                            "fontStyle": "normal",
                                            "fontWeight": "normal",
                                          },
                                          Object {},
                                        ]
                                      }
                                      testID="iconIcon"
                                    >
                                      
                                    </Text>
                                  </View>
                                </View>
                              </View>
                            </View>
                          </View>
                          <View
                            style={
                              Array [
                                Object {
                                  "alignItems": "center",
                                  "flexDirection": "row",
                                  "justifyContent": "center",
                                },
                              ]
                            }
                          >
                            <View
                              accessible={true}
                              focusable={true}
                              onClick={[Function]}
                              onResponderGrant={[Function]}
                              onResponderMove={[Function]}
                              onResponderRelease={[Function]}
                              onResponderTerminate={[Function]}
                              onResponderTerminationRequest={[Function]}
                              onStartShouldSetResponder={[Function]}
                              style={
                                Object {
                                  "opacity": 1,
                                }
                              }
                            >
                              <View
                                style={
                                  Object {
                                    "overflow": "hidden",
                                  }
                                }
                              >
                                <View>
                                  <View
                                    style={
                                      Object {
                                        "alignItems": "center",
                                        "backgroundColor": "transparent",
                                        "justifyContent": "center",
                                      }
                                    }
                                  >
                                    <Text
                                      allowFontScaling={false}
                                      style={
                                        Array [
                                          Object {
                                            "color": "#fff",
                                            "fontSize": 30,
                                          },
                                          Object {
                                            "backgroundColor": "transparent",
                                          },
                                          Object {
                                            "fontFamily": "Material Icons",
                                            "fontStyle": "normal",
                                            "fontWeight": "normal",
                                          },
                                          Object {},
                                        ]
                                      }
                                      testID="iconIcon"
                                    >
                                      
                                    </Text>
                                  </View>
                                </View>
                              </View>
                            </View>
                            <View
                              style={
                                Object {
                                  "overflow": "hidden",
                                }
                              }
                            >
                              <View
                                accessible={true}
                                focusable={true}
                                onClick={[Function]}
                                onResponderGrant={[Function]}
                                onResponderMove={[Function]}
                                onResponderRelease={[Function]}
                                onResponderTerminate={[Function]}
                                onResponderTerminationRequest={[Function]}
                                onStartShouldSetResponder={[Function]}
                                style={
                                  Array [
                                    Object {
                                      "marginBottom": 0,
                                      "marginLeft": 32,
                                      "marginRight": 32,
                                      "marginTop": 0,
                                    },
                                  ]
                                }
                              >
                                <View
                                  style={
                                    Object {
                                      "alignItems": "center",
                                      "backgroundColor": "transparent",
                                      "justifyContent": "center",
                                    }
                                  }
                                >
                                  <Text
                                    allowFontScaling={false}
                                    style={
                                      Array [
                                        Object {
                                          "color": "#ffffff",
                                          "fontSize": 52,
                                        },
                                        Object {
                                          "backgroundColor": "transparent",
                                        },
                                        Object {
                                          "fontFamily": "Material Icons",
                                          "fontStyle": "normal",
                                          "fontWeight": "normal",
                                        },
                                        Object {},
                                      ]
                                    }
                                    testID="iconIcon"
                                  >
                                    
                                  </Text>
                                </View>
                              </View>
                            </View>
                            <View
                              accessible={true}
                              focusable={true}
                              onClick={[Function]}
                              onResponderGrant={[Function]}
                              onResponderMove={[Function]}
                              onResponderRelease={[Function]}
                              onResponderTerminate={[Function]}
                              onResponderTerminationRequest={[Function]}
                              onStartShouldSetResponder={[Function]}
                              style={
                                Object {
                                  "opacity": 1,
                                }
                              }
                            >
                              <View
                                style={
                                  Object {
                                    "overflow": "hidden",
                                  }
                                }
                              >
                                <View>
                                  <View
                                    style={
                                      Object {
                                        "alignItems": "center",
                                        "backgroundColor": "transparent",
                                        "justifyContent": "center",
                                      }
                                    }
                                  >
                                    <Text
                                      allowFontScaling={false}
                                      style={
                                        Array [
                                          Object {
                                            "color": "#fff",
                                            "fontSize": 30,
                                          },
                                          Object {
                                            "backgroundColor": "transparent",
                                          },
                                          Object {
                                            "fontFamily": "Material Icons",
                                            "fontStyle": "normal",
                                            "fontWeight": "normal",
                                          },
                                          Object {},
                                        ]
                                      }
                                      testID="iconIcon"
                                    >
                                      
                                    </Text>
                                  </View>
                                </View>
                              </View>
                            </View>
                          </View>
                          <View
                            onStartShouldSetResponder={[Function]}
                            onTouchEnd={[Function]}
                            style={
                              Array [
                                Object {
                                  "alignItems": "center",
                                  "flexDirection": "row",
                                  "height": 48,
                                  "justifyContent": "center",
                                  "marginBottom": 24,
                                  "marginLeft": 48,
                                  "marginRight": 48,
                                  "marginTop": 0,
                                  "paddingBottom": 0,
                                  "paddingLeft": 10,
                                  "paddingRight": 10,
                                  "paddingTop": 0,
                                },
                              ]
                            }
                          >
                            <View
                              collapsable={false}
                              onGestureHandlerEvent={[Function]}
                              onGestureHandlerStateChange={[Function]}
                              style={
                                Object {
                                  "width": 600,
                                }
                              }
                            >
                              <View
                                animated={true}
                                animationConfig={
                                  Object {
                                    "bounciness": 0,
                                  }
                                }
                                animationType="spring"
                                indeterminate={false}
                                indeterminateAnimationDuration={1000}
                                onLayout={[Function]}
                                progress={0}
                                style={
                                  Array [
                                    Object {
                                      "backgroundColor": "rgba(255, 255, 255, 0.5)",
                                      "borderColor": "white",
                                      "borderRadius": 4,
                                      "borderWidth": 1,
                                      "overflow": "hidden",
                                      "width": 600,
                                    },
                                    Array [
                                      Object {
                                        "borderColor": "rgba(255, 255, 255, 0)",
                                        "color": "#fff",
                                        "height": 3,
                                      },
                                    ],
                                  ]
                                }
                                useNativeDriver={false}
                              >
                                <View
                                  style={
                                    Object {
                                      "backgroundColor": "white",
                                      "height": 6,
                                      "transform": Array [
                                        Object {
                                          "translateX": -2.842170943040401e-14,
                                        },
                                        Object {
                                          "translateX": -299,
                                        },
                                        Object {
                                          "scaleX": 0.0001,
                                        },
                                      ],
                                    }
                                  }
                                />
                              </View>
                              <View
                                style={
                                  Object {
                                    "bottom": -8,
                                    "flexDirection": "column",
                                    "position": "absolute",
                                    "transform": Array [
                                      Object {
                                        "translateX": 0,
                                      },
                                    ],
                                    "width": 600,
                                  }
                                }
                              >
                                <Text
                                  style={
                                    Array [
                                      Object {
                                        "color": "#ffffff",
                                        "fontSize": 14,
                                        "fontWeight": "700",
                                        "left": -8,
                                      },
                                    ]
                                  }
                                >
                                  00:00
                                </Text>
                                <View
                                  style={
                                    Array [
                                      Object {
                                        "backgroundColor": "#fff",
                                        "borderRadius": 9,
                                        "height": 18,
                                        "width": 18,
                                      },
                                    ]
                                  }
                                />
                              </View>
                            </View>
                            <Text
                              style={
                                Array [
                                  Object {
                                    "color": "#fff",
                                    "fontSize": 14,
                                    "fontWeight": "700",
                                    "marginLeft": 15,
                                    "opacity": 0.5,
                                  },
                                ]
                              }
                            >
                              00:00
                            </Text>
                          </View>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </RNSScreen>
    </RNSScreenContainer>
    <View
      pointerEvents="box-none"
      style={
        Object {
          "height": 64,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "zIndex": 1,
        }
      }
    />
  </RNCSafeAreaProvider>
</View>
`;
